cloudinit/sources/DataSourceWSL.py

```diff
@@ -157,35 +163,73 @@ def __init__(self, path: PurePath):
         if "text/cloud-config" == type_from_starts_with(self.raw):
             self.config_dict = util.load_yaml(self.raw)
 
-    def is_cloud_config(self) -> bool:
-        return self.config_dict is not None
+        if (
+            self.config_dict  # Valid non-empty config
+            and instance_id
+            and instance_id != DEFAULT_INSTANCE_ID  # custom instance-id
+            and self.config_dict.get(LANDSCAPE_CFG_KEY, {}).get(
+                LANDSCAPE_CLIENT_CFG_KEY
+            )
+            is not None  # Landscape client config exists
+        ):
+            # Let's set the installation_request_id to the metadata.instance-id
+            # if not already set, and update the raw config if we modified it:
+            req_id = self.config_dict[LANDSCAPE_CFG_KEY][
+                LANDSCAPE_CLIENT_CFG_KEY
+            ].setdefault(LANDSCAPE_INSTALLATION_REQ_ID, instance_id)
+            if req_id == instance_id:
+                self.raw = (
+                    "#cloud-config\n%s" % yaml.dump(self.config_dict).strip()
+                )
 
 
-def load_instance_metadata(
-    cloudinitdir: Optional[PurePath], instance_name: str
-) -> dict:
+def _load_metadata(metadata_path: PurePath) -> Optional[dict]:
     """
-    Returns the relevant metadata loaded from cloudinit dir based on the
-    instance name
+    Returns the relevant metadata loaded from the supplied metadata_path.
     """
-    metadata = {"instance-id": DEFAULT_INSTANCE_ID}
-    if cloudinitdir is None:
-        return metadata
-    metadata_path = os.path.join(
-        cloudinitdir.as_posix(), "%s.meta-data" % instance_name
-    )
-
+    metadata = None
     try:
-        metadata = util.load_yaml(util.load_text_file(metadata_path))
+        metadata = util.load_yaml(
+            util.load_text_file(metadata_path), default={}
+        )
     except FileNotFoundError:
         LOG.debug(
-            "No instance metadata found at %s. Using default instance-id.",
+            "No instance metadata found at %s.",
             metadata_path,
         )
+
+    return metadata
+
+
+def load_instance_metadata(user_home: PurePath, instance_name: str) -> dict:
+    """
+    Returns the relevant metadata loaded from either the Ubuntu Pro dir
+    or the user cloud-int dir based on the instance name
+    """
+    metadata = {"instance-id": DEFAULT_INSTANCE_ID}
+    pro_dir = cloud_init_data_dir(user_home / ".ubuntupro")
```

Comment:
Should this actually be the following to align with docs and ds-identify behavior?
```suggestion
    pro_dir = cloud_init_data_dir(user_home / ".ubuntupro/.cloud-init")
```